### ISSUE TITLE
" There is no route defined for key " Problem

### ISSUE THREAD

**Issue by alirezadarvishi** [react-navigation/react-navigation#6116]
I want to go to Cart Screen from product screen when I'm in Men's tab.

home.js >> Men's tab >> Product screen >> Cart Screen : Error

> Error: There is no route defined for key Cart. Must be one of: 'menStackNav','Products'

My codes in snack: https://snack.expo.io/@alirezadarvishi/full-screen-modal-v3

> Goal: I want to go cart Screen, When I click on cart screen Button in men's tab.

**Comment by alirezadarvishi** [comment]
@matthamil  Can you hlep me?

**Comment by vonovak** [comment]
hi. The repro you provided is kinda hard to follow. The problem is that you're calling `navigate` in a stack that does not have the screen in it. You can try using `reset` or maybe `dangerouslyGetParent` to get hold of the correct navigator, take a look at https://reactnavigation.org/docs/en/navigation-prop.html#dangerouslygetparent-get-parent-navigator
